### PR TITLE
Level attribute handling fixup

### PR DESCRIPTION
**Pull request by hcarter-775**

### CODE REVIEW

File: drivers/SmartThings/matter-pump/src/init.lua [hunk 1]
```diff
@@ -176,7 +176,8 @@ end
 
 local function level_attr_handler(driver, device, ib, response)
   if ib.data.value ~= nil then
-    local level = math.floor((ib.data.value / 254.0 * 100) + 0.5)
+    local level = math.floor((ib.data.value / 200 * 100) + 0.5)
```
Comment (ctowns) [inline drivers/SmartThings/matter-pump/src/init.lua:179]: Could you define these magic numbers at the top of the file?

```
-- Per matter spec, the pump level is in steps of 0.5% and the max level value is 200. Anything above is considered 100%
local MAX_PUMP_ATTR_LEVEL = 200
local MAX_LEVEL = 100
```
And then change this to something like
```suggestion
    local level = math.floor((ib.data.value / MAX_PUMP_ATTR_LEVEL * MAX_LEVEL) + 0.5)
    level = math.max(level, MAX_LEVEL)
```

Comment (hcarter-775) [reply drivers/SmartThings/matter-pump/src/init.lua]: done :+1:

File: drivers/SmartThings/matter-pump/src/init.lua [hunk 2]
```diff
@@ -176,7 +180,8 @@ end
 
 local function level_attr_handler(driver, device, ib, response)
   if ib.data.value ~= nil then
-    local level = math.floor((ib.data.value / 254.0 * 100) + 0.5)
+    local level = math.floor((ib.data.value / MAX_PUMP_ATTR_LEVEL * MAX_CAP_SWITCH_LEVEL) + 0.5)
+    level = math.max(level, MAX_CAP_SWITCH_LEVEL)
```
Comment (ctowns) [inline drivers/SmartThings/matter-pump/src/init.lua:184]: Whoops, sorry this should be math.min :)

Comment (hcarter-775) [reply drivers/SmartThings/matter-pump/src/init.lua:184]: of course, duh! good catch

**Comment by hcarter-775** [comment]
tested with a partner pump device and things are working smoothly

**Review by ctowns — APPROVED** [review]

**Pull request merged**